pages/account/statistics.js

```diff
@@ -31,9 +32,17 @@ export async function getServerSideProps(context) {
       },
     };
   }
+  let account = await getAccountByProviderAccountId(session.user.id);
+
   const username = session.username;
   const { status, profile } = await getUserApi(req, res, username);
-  if (status !== 200) {
+
+  if (status === 200) {
+    //Linking a profile to an account is a one-time process. Subsequent logins will disregard any attempt to add the profile again.
+    if (account) {
+      await associateProfileWithAccountIfAbsentAndReturnAccount(account, profile._id);
```

Comment:
Thinking about it, could this be triggered on login instead of the stats page?

Reply:
Let me come back on this.

Reply:
After playing I managed to learn more about next-auth and do this when signing in, so I merged your PR into a temporary branch and  made my changes, you can see both our changes here https://github.com/EddieHubCommunity/LinkFree/pull/8315